python/paddle/distributed/auto_parallel/attribute.py

```diff
@@ -0,0 +1,307 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+import copy
+from collections import defaultdict
+
+
+class TensorDistributedAttribute:
+    def __init__(self, owner_tensor, owner_context):
+        self._owner_tensor = owner_tensor
+        self._owner_context = owner_context
+        self._process_mesh = None
+        self._dims_mapping = None
+        self._shard_mask = None
+        self._offload_device = None
+        self._shape = None
+        self._is_annotated = {}
+        self._is_parameter = False
+
+    def get_owner_tensor(self):
+        return self._owner_tensor
+
+    def get_owner_context(self):
+        return self._owner_context
+
+    def get_process_mesh(self):
+        return self._process_mesh
+
+    def set_process_mesh(self, process_mesh):
+        self._process_mesh = process_mesh
+
+    def get_dims_mapping(self):
```

Comment:
Lots of implementations relay on dims_mapping. Since this isn't exposed to user, I left it alone. Besides, I think we should try to convince @XiaoguangHu01  to adopt dims_mapping as tf.